frame/scheduler/src/lib.rs

```diff
@@ -96,24 +102,24 @@ struct ScheduledV1<Call, BlockNumber> {
 
 /// Information regarding an item to be executed in the future.
 #[cfg_attr(any(feature = "std", test), derive(PartialEq, Eq))]
-#[derive(Clone, RuntimeDebug, Encode, Decode, TypeInfo)]
-pub struct ScheduledV3<Call, BlockNumber, PalletsOrigin, AccountId> {
+#[derive(Clone, RuntimeDebug, Encode, Decode, MaxEncodedLen, TypeInfo)]
+pub struct Scheduled<Call, BlockNumber, PalletsOrigin, AccountId> {
 	/// The unique identity for this task, if there is one.
-	maybe_id: Option<Vec<u8>>,
+	maybe_id: Option<[u8; 32]>,
```

Comment:
The scheduler migration fails in `try-runtime` on Kusama over here https://github.com/paritytech/polkadot/pull/6018.
Errors with `frame_support::storage::generator::map: Invalid translate: fail to decode old value`.  

Could be this change maybe? I will swap it for a generic type and try again.

Reply:
this would indeed fail.